File: backend/src/main/java/lan/dk/podcastserver/service/health/DownloaderHealthIndicator.java
package lan.dk.podcastserver.service.health;

import lan.dk.podcastserver.manager.ItemDownloadManager;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.actuate.health.AbstractHealthIndicator;
import org.springframework.boot.actuate.health.Health;
import org.springframework.stereotype.Component;

/**
 * Created by kevin on 19/11/2017
 */
@Component
@RequiredArgsConstructor
public class DownloaderHealthIndicator extends AbstractHealthIndicator {

    private final ItemDownloadManager itemDownloadManager;

    @Override
    protected void doHealthCheck(Health.Builder builder) throws Exception {
        // @formatter:off
        builder.up()
            .withDetail("isDownloading", itemDownloadManager.getNumberOfCurrentDownload() > 0)
            .withDetail("numberOfParallelDownloads", itemDownloadManager.getLimitParallelDownload())
            .withDetail("numberOfDownloading", itemDownloadManager.getNumberOfCurrentDownload())
            .withDetail("downloadingItems", itemDownloadManager.getItemsInDownloadingQueue())
            .withDetail("numberInQueue", itemDownloadManager.getWaitingQueue().length())
            .withDetail("waitingItems", itemDownloadManager.getWaitingQueue())
        .build();
        // @formatter:on
    }

}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/service/health/DownloaderHealthIndicator.kt
package com.github.davinkevin.podcastserver.service.health

import com.github.davinkevin.podcastserver.manager.ItemDownloadManager
import org.springframework.boot.actuate.health.AbstractHealthIndicator
import org.springframework.boot.actuate.health.Health
import org.springframework.stereotype.Component

/**
 * Created by kevin on 19/11/2017
 */
@Component
class DownloaderHealthIndicator(val itemDownloadManager: ItemDownloadManager) : AbstractHealthIndicator() {

    override fun doHealthCheck(builder: Health.Builder) {
        val waiting = itemDownloadManager.waitingQueue
        val items = itemDownloadManager.downloadingItems

        // @formatter:off
        builder.up()
            .withDetail("isDownloading", items.isNotEmpty())
            .withDetail("numberOfParallelDownloads", itemDownloadManager.limitParallelDownload)
            .withDetail("numberOfDownloading", items.size)
            .withDetail("downloadingItems", items)
            .withDetail("numberInQueue", waiting.size)
            .withDetail("waitingItems", waiting)
        .build()
        // @formatter:on
    }

}
